routers/web/user/auth.go

```diff
@@ -1429,6 +1429,10 @@ func handleAccountActivation(ctx *context.Context, user *models.User) {
 		return
 	}
 
+	if err := models.ActivateUserEmail(user.ID, user.Email, true); err != nil {
+		log.Error(fmt.Sprintf("Error active user mail: %v", err))
+	}
+
```

Comment:
I have no idea that whether primary email is required in any cases when calling `handleAccountActivation`.  
If primary email required anyway,
```suggestion
	if err := models.ActivateUserEmail(user.ID, user.Email, true); err != nil {
	  ctx.ServerError("ActivateUserEmail", err)
		log.Error(fmt.Sprintf("Error active user mail: %v", err))
		return
	}
```
`return` should be added.
Or, the `handleAccountActivation` could be:
```go
func handleAccountActivation(ctx *context.Context, user *models.User) {
	if err := models.ActivateUserEmail(user.ID, user.Email, true); err != nil {
		if models.IsErrUserNotExist(err) {
			ctx.NotFound("UpdateUserCols", err)
		} else {
			ctx.ServerError("AccountActivation", err)
		}
		log.Error(fmt.Sprintf("Error active user mail: %v", err))
		return
	}

	user.IsActive = true
	log.Trace("User activated: %s", user.Name)

	if err := ctx.Session.Set("uid", user.ID); err != nil {
		log.Error(fmt.Sprintf("Error setting uid in session: %v", err))
	}
	if err := ctx.Session.Set("uname", user.Name); err != nil {
		log.Error(fmt.Sprintf("Error setting uname in session: %v", err))
	}
	if err := ctx.Session.Release(); err != nil {
		log.Error("Error storing session: %v", err)
	}

	ctx.Flash.Success(ctx.Tr("auth.account_activated"))
	ctx.Redirect(setting.AppSubURL + "/")
}
```
, for the `ActivateUserEmail` will activate user account if the email address is primary.

Reply:
have applied the change above